packages/kubernetes-1.15/kubelet-config

```diff
@@ -29,6 +29,34 @@ authorization:
 clusterDomain: {{settings.kubernetes.cluster-domain}}
 clusterDNS:
 - {{settings.kubernetes.cluster-dns-ip}}
+{{~#if settings.kubernetes.eviction-hard}}
+evictionHard:
+  {{~#each settings.kubernetes.eviction-hard}}
+  {{@key}}: "{{this}}"
+  {{~/each}}
+{{~/if}}
+{{~#if settings.kubernetes.allowed-unsafe-sysctls}}
+allowedUnsafeSysctls: {{settings.kubernetes.allowed-unsafe-sysctls}}
+{{~/if}}
+kubeReserved:
+  {{~#if settings.kubernetes.kube-reserved.cpu}}
+  cpu: "{{settings.kubernetes.kube-reserved.cpu}}"
+  {{~else}}
+  cpu: "{{get_cpu_millicores_to_reserve}}"
```

Comment:
Instead of an if/else here, should we consider putting the switching logic in the Schnauzer helper instead? So Schnauzer either use the supplied argument `settings.kubernetes.kube-reserved.cpu` or calculate the backup value if it is empty.

I'm assuming that `get_cpu_millicores_to_reserve  settings.kubernetes.kube-reserved.cpu` would *not* error if the user did not set `kube-reserved.cpu`.

It would cut down on the unsightliness of the template file but I'm not 100% sure it's a better solution. Do others have thoughts?

Reply:
I'm also wondering why one of the names is `get_` and the other is `calculate_`

Reply:
Oh I have a function in `Schnauzer ` named `calculate_cpu_..`, so I use `get_` here. I think I should change it to `calculate_` to be more consistent.

Reply:
> I'm assuming that get_cpu_millicores_to_reserve settings.kubernetes.kube-reserved.cpu would not error if the user did not set kube-reserved.cpu.

I think we can treat it like the `default` helper. 

https://github.com/bottlerocket-os/bottlerocket/blob/513c8a2033a7dee90a31061c9f67895e5b685b5d/sources/api/schnauzer/src/helpers.rs#L411-L418

Instead of the `default.to_string()`, it would be something like `calculate_cpu_reserve(num_cores).to_string();`